airflow/www/views.py

```diff
@@ -462,6 +462,21 @@ def chart(self):
             label=chart.label,
             embed=embed)
 
+    @expose('/task_info')
+    def task_info(self):
+        """Returns a JSON with a task's public instance variables. """
+        dag_id = request.args.get('dag_id')
+        task_id = request.args.get('task_id')
+        if dag_id not in dagbag.dags:
+            return wwwutils.json_response({})
+        tasks = [task for task in dagbag.dags[dag_id].tasks if task.task_id == task_id]
```

Comment:
Breaking dag into a separate variable reduces nesting and makes the code clearer, also you can just use get_task instead of the list comprehension:

dag = dagbag.dags[dag_id]
dag.get_task(task_id)